src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -66,7 +65,7 @@
      * if your step cannot trivially handle a null value of a given kind, list that type here.
      * The Pipeline execution engine will then signal a user error before even starting your step if called in an inappropriate context.
      * For example, a step requesting a {@link Launcher} may only be run inside a {@code node {…}} block.
-     * @return typically an {@link ImmutableSet#of(Object)} with context types like {@link TaskListener} or {@link Run} or {@link FilePath}
+     * @return typically a {@link Collections#unmodifiableSet(Set)} with context types like {@link TaskListener} or {@link Run} or {@link FilePath}
```

Comment:
I actually recommended `ImmutableSet.of` because it lets you return multiple values in a one-liner, not because of the immutability per se.
```suggestion
     * @return a set of context types like {@link TaskListener} or {@link Run} or {@link FilePath}
```

Reply:
I guess with Java 11 we can use `Set.of` for that purpose.